### PR TITLE
Update documentation for v1

### LINKED ISSUE #467: Update documentation
- [x] Add architecture design information.
- [ ] Add Macroscope section to configure crawlers.
- [ ] Add group definition information.

### PR DESCRIPTION
**Pull request by TristanCacqueray**
Fixes #467 https://github.com/change-metrics/monocle/issues/494

### CODE REVIEW

File: docker-compose.dhall [hunk 1]
```diff
@@ -185,6 +185,35 @@ let createApiNgService =
                                          }
                                    )
 
+let createCrawlerNgService =
```
Comment (georgettica) [inline docker-compose.dhall:188]:
```suggestion
let createCrawlerService =
```

File: .github/workflows/publish-master.yaml [hunk 2]
```diff
@@ -58,17 +58,17 @@ jobs:
 
       - name: Check api service
```
Comment (morucci) [inline .github/workflows/publish-master.yaml:59]: Check api-legacy service

File: .github/workflows/publish-master.yaml [hunk 3]
```diff
@@ -58,17 +58,17 @@ jobs:
...
           curl -s http://localhost:9876/api/0/indices
 
       - name: Check api-ng server
-        run: docker-compose logs api-ng
+        run: docker-compose logs api
 
       - name: Check web server
         run: docker-compose logs web
 
       - name: Check crawler service
```
Comment (morucci) [inline .github/workflows/publish-master.yaml:70]: Check crawler-legacy service

File: README.md [hunk 4]
```diff
@@ -9,23 +9,25 @@ join us in the Matrix room at [#monocle:matrix.org](https://matrix.to/#/#monocle
...
 
-To summarize we have the following components in Monocle:
+Monocle is composed of the following services:
 
 1. an Elasticsearch data store.
-2. an api service.
-3. a crawler service.
-4. a web UI.
+2. an api service to perform user query and index crawler output.
+3. a crawler service to retrive change from provider.
```
Comment (morucci) [inline README.md:21]: retrieve

File: README.md [hunk 5]
```diff
@@ -9,23 +9,25 @@ join us in the Matrix room at [#monocle:matrix.org](https://matrix.to/#/#monocle
...
+4. a web proxy and single page web application.
```
Comment (morucci) [inline README.md:22]: a web proxy and web application to browse metrics

File: README.md [hunk 6]
```diff
@@ -9,23 +9,25 @@ join us in the Matrix room at [#monocle:matrix.org](https://matrix.to/#/#monocle
...
+
+The APIs are defined using [protobuf][monocle-protobuf] and served over HTTP through [Monocle OpenAPI][monocle-openapi].
+
+Some legacy component are still required until they are migrated to the new OpenAPI (see the related issues):
+
+5. a api service to perform filter based query [issue 468](https://github.com/change-metrics/monocle/issues/468).
```
Comment (morucci) [inline README.md:28]: an api ...

File: README.md [hunk 7]
```diff
@@ -84,25 +89,22 @@ Ensure services are running:
...
 ```
 
 You should be able to access the web UI at <http://localhost:8080>.
 
 After a change in the configuration file, the api and crawler services need to be restarted:
 
 ```ShellSession
-$ docker-compose restart api
-$ docker-compose restart crawler
+$ docker-compose restart api-legacy
```
Comment (morucci) [inline README.md:106]: docker-compose restart api

The new api still need to be restarted. Better to add it as of now.

**Review by morucci — CHANGES_REQUESTED** [review]
Some comment - still I need to test a docker deployment.
Also either we remove or update the contrib/start-with-podman.sh script.

**Comment by morucci** [comment]
docker-compose build fails on:
Step 3/5 : RUN cabal v2-install -v1 exe:monocle-api exe:macroscope
 ---> Running in 22599231f538
cabal: filepath wildcard 'dhall-monocle/Monocle/**/*.dhall' refers to the
directory 'dhall-monocle/Monocle', which does not exist or is not a directory.

**Comment by TristanCacqueray** [comment]
> directory 'dhall-monocle/Monocle', which does not exist or is not a directory.

This indicate the submodule is not checked outu.

File: README.md (hunk 7 above)
Comment (TristanCacqueray) [reply README.md:107]: Let's fix that with https://github.com/change-metrics/monocle/issues/494 instead

File: haskell/app/Macroscope.hs [hunk 8]
```diff
@@ -22,9 +22,10 @@ instance ParseRecord (Macroscope Wrapped) where
 main :: IO ()
 main = do
   args <- unwrapRecord "Macroscope lentille runner"
-  withClient (monocleUrl args) Nothing $ \client ->
+  config' <- fromMaybe (error "--config or CONFIG env is required") <$> lookupEnv "CONFIG"
+  withClient (fromMaybe "http://api-ng:9898" $ monocleUrl args) Nothing $ \client ->
```
Comment (morucci) [inline haskell/app/Macroscope.hs:26]: http://api:9898

**Review by morucci — APPROVED** [review]
Ok the macroscope service starts part of a docker-compose deployment.
Just a weird behavior where after the first loop it seems stuck. I was expecting to see the loop triggered every 600s. But still nothing more happen. Anyway I'll merge this PR.

**Pull request merged**